src/main/java/com/adobe/target/edge/client/http/DefaultTargetHttpClient.java

```diff
@@ -130,14 +136,41 @@ public void addDefaultHeader(String key, String value) {
     return rawResponse -> {
       TimingTool timer = new TimingTool();
       timer.timeStart(TIMING_EXECUTE_REQUEST);
-      String rawResponseContent = rawResponse.getContentAsString();
+      byte[] responseAsByte = rawResponse.getContentAsBytes();
+      String rawResponseContent;
+      try {
+        rawResponseContent = new String(responseAsByte, getCharSet(rawResponse));
+      } catch (UnsupportedEncodingException e) {
+        throw new UnirestException(e);
+      }
       R responseBody = getObjectMapper().readValue(rawResponseContent, responseClass);
       responseWrapper.setParsingTime(timer.timeEnd(TIMING_EXECUTE_REQUEST));
-      responseWrapper.setResponseSize(rawResponseContent.length());
+      responseWrapper.setResponseSize(responseAsByte.length);
       return responseBody;
     };
   }
 
+  protected String getCharSet(RawResponse rawResponse) {
```

Comment:
Implementation looks good.  Were there any unit tests from the unirest lib that cover these methods?  If so, let's copy them over.

Reply:
DefaultTargetHttpClient class is not covered with good coverage yet. Let me create a small ticket to cover test coverage for delivery API call. Since the methods I have added is private, it will be resolved as part of that.

Reply:
https://jira.corp.adobe.com/browse/TNT-42930